packages/wallet/ui/src/util/SuggestChain.js

```diff
@@ -5,36 +5,33 @@ export const AGORIC_COIN_TYPE = 564;
 export const COSMOS_COIN_TYPE = 118;
 
 /**
- * @param {string} networkConfig URL
- * @param {object} addrs
- * @param {string} addrs.rpcAddr URL or origin
- * @param {string} addrs.apiAddr URL
- * @param {string} chainId
- * @param {string} [caption]
+ * @param {object} io
+ * @param {typeof Math.random} io.random
+ * @param {import('@agoric/internal').NetworkConfig} networkConfig
+ * @param {string} caption
  * @returns {import('@keplr-wallet/types').ChainInfo}
  */
-const makeChainInfo = (
-  networkConfig,
-  { rpcAddr, apiAddr },
-  chainId,
-  caption,
-) => {
-  const coinType = Number(
-    new URL(networkConfig).searchParams.get('coinType') || AGORIC_COIN_TYPE,
-  );
-  const hostname = new URL(networkConfig).hostname;
-  const network = hostname.split('.')[0];
+export const makeChainInfo = (io, networkConfig, caption) => {
+  const { chainName, rpcAddrs, apiAddrs } = networkConfig;
+  const index = Math.floor(io.random() * rpcAddrs.length);
+
+  const rpcAddr = rpcAddrs[index];
   const rpc = rpcAddr.match(/:\/\//) ? rpcAddr : `http://${rpcAddr}`;
 
+  const rest = apiAddrs
+    ? // pick the same index
+      apiAddrs[index]
+    : // adapt from rpc
+      rpc.replace(/(:\d+)?$/, ':1317');
+
   return {
     rpc,
-    rest: apiAddr,
-    chainId,
-    chainName: caption || `Agoric ${network}`,
+    rest,
+    chainId: chainName,
+    chainName: caption,
     stakeCurrency,
-    walletUrlForStaking: `https://${network}.staking.agoric.app`,
```

Comment:
No more staking link? That looks like a regression / breaking change. I have used it from time to time. I suppose I won't miss it that badly.

Reply:
Hm... we did ship, so maybe it's worth more discipline about this sort of thing. For the benefit of release notes, how about a separate BREAKING CHANGE commit to take out `walletUrlForStaking` before this `feat`?

Reply:
I found it nowhere in the Agoric org (https://github.com/Agoric/agoric-sdk/search?q=walletUrlForStaking) and I didn't think it was really part of the API of this util.

It's technically breaking the API, but I don't think it breaks anything. If we're going to get disciplined about breaking changes, then I think we also need to get more disciplined about what we export.

In any case, I'll try separating this out as a BREAKING CHANGE to see how it feels ;-)

Reply:
It's not used in our code; it's used in by Keplr. Again: this is a network protocol / API.

Reply:
Ah. I'll restore it.

(Fwiw, I don't see the antecedent for "again". It definitely wasn't clear from your earlier comments that this was an actual regression. )